runbotics/runbotics-orchestrator-ui/src/components/ResizableDrawer/ResizableDrawer.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+import styled from 'styled-components';
+import { Drawer } from '@mui/material';
+import { DrawerProps } from '@mui/material/Drawer/Drawer';
+import useWrappedState from 'src/hooks/useWrappedState';
+
+const INITIAL_WIDTH = 300;
+const MIN_WIDTH = 250;
+const MAX_WIDTH = 600;
+
+const Dragger = styled('div')(({ theme }) => ({
+    width: '3px',
+    cursor: 'ew-resize',
+    pointerEvents: 'all',
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    bottom: 0,
+
+    ':hover, [data-dragmode] &': {
+        backgroundColor: `${theme.palette.primary.light}`,
+    },
+}));
+
+const ResizableDrawer = ({ children, open, ...other }: DrawerProps) => {
+    const [width, setWidth] = useWrappedState(INITIAL_WIDTH, { min: MIN_WIDTH, max: MAX_WIDTH });
+
+    const handleMousemove = (event: MouseEvent) => {
+        setWidth(window.innerWidth - event.clientX);
+    };
+
+    const handleDragEnd = () => {
+        document.removeEventListener('mousemove', handleMousemove);
```

Comment:
these handlers are not written in react way, I assume that it is legacy. Lets change it